### PR TITLE
Create photo gallery page

### PR DESCRIPTION
**Pull request by andrevieira11**

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -1,24 +1,129 @@
-import Head from 'next/head'
-import Image from 'next/image'
-import styles from '../styles/Home.module.css'
+import Head from "next/head";
+import Image from "next/image";
+import styles from "../styles/Home.module.css";
+import { useState } from "react";
+
+const DECODE_PREFIX = "data:image/png;base64,";
+const API = "http://127.0.0.1:8000/photo/";
```
Comment (hugo-serra) [inline pages/index.js:7]: put this in an env file

**Review by hugo-serra — CHANGES_REQUESTED** [review]

File: pages/contest/[id].js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    }
+  }
+
+  function changeDescription(event) {
+    setSubmissionDescription(event.target.value);
+  }
+
+  async function uploadToServer() {
+    const body = new FormData();
+    body.append("user", "5d2ee9f3-7f36-4861-ad59-4297aa96f932");
```
Comment (hugo-serra) [inline pages/contest/[id].js:39]: Add a TODO here to refactor when we have auth

File: pages/contest/[id].js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  return {
+    props: {
+      contestId,
+      submissions,
+      userList,
+    },
+  };
+}
+
+function BlurImage({ contestId, submissionId, url, user, description }) {
```
Comment (hugo-serra) [inline pages/contest/[id].js:117]: This seems like a Component, let's create a components folder and place all the components inside

File: pages/index.js [hunk 4]
```diff
@@ -1,24 +1,123 @@
...
+import Image from "next/image";
+import { useState } from "react";
+import styles from "../styles/Home.module.css";
+const { NEXT_PUBLIC_GET, NEXT_PUBLIC_POST } = process.env;
+
+import { fetchEndpoint } from "../services/utils.js";
+
+const DECODE_PREFIX = "data:image/png;base64,";
+
+async function getSubmissionList() {
```
Comment (hugo-serra) [inline pages/index.js:11]: This should be in a service

File: pages/index.js [hunk 5]
```diff
@@ -1,24 +1,123 @@
...
+  return fetchEndpoint("submissions/", NEXT_PUBLIC_GET);
+}
+
+async function getUserList() {
```
Comment (hugo-serra) [inline pages/index.js:15]: This should be in a service

File: pages/index.js [hunk 6]
```diff
@@ -1,24 +1,123 @@
...
+
+      reader.onloadend = () => {
+        setCreateObjectURL(reader.result);
+      };
+
+      reader.readAsDataURL(i);
+    }
+  }
+
+  async function uploadToServer() {
```
Comment (hugo-serra) [inline pages/index.js:80]: This should be in a service with a better name

File: services/utils.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
+const NEXT_PUBLIC_API_URL = process.env.NEXT_PUBLIC_API_URL;
+const NEXT_PUBLIC_GET = process.env.NEXT_PUBLIC_GET;
```
Comment (hugo-serra) [inline services/utils.js:2]: Get and post dont need to be environment variables. We can use class constants here

File: services/utils.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
+const NEXT_PUBLIC_API_URL = process.env.NEXT_PUBLIC_API_URL;
```
Comment (hugo-serra) [inline services/utils.js:1]: This file is a service for what? It should be named after that.

**Review by hugo-serra — CHANGES_REQUESTED** [review]

File: pages/contest/[id].js [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+        </div>
+      </main>
+      <footer className={styles.footer}></footer>
+    </div>
+  );
+}
+
+export async function getServerSideProps(context) {
+  let contestId = context.params.id;
+  let submissions;
```
Comment (hugo-serra) [inline pages/contest/[id].js:92]: this should probably have a default value

Comment (hugo-serra) [reply pages/contest/[id].js:92]: as well as the others

**Review by hugo-serra — APPROVED** [review]

**Review by JoaoCVerissimo — APPROVED** [review]

**Pull request merged**